typescript-recommended.js

```diff
@@ -10,22 +10,97 @@ module.exports = {
   parserOptions: {
     project: './tsconfig.json',
     tsconfigRootDir: '.',
-    ecmaVersion: 2018,
+    ecmaVersion: 2020,
     sourceType: 'module',
   },
   rules: {
+    'prettier/prettier': [
+      'error',
+      {
+        bracketSpacing: true,
+        trailingComma: 'all',
+        singleQuote: true,
+        semi: true,
+        useTabs: false,
+        tabWidth: 2,
+        printWidth: 120,
+      },
+    ],
     eqeqeq: 'error',
     curly: 'error',
+    'no-trailing-spaces': 'error',
+    'eol-last': ['error', 'always'],
     'prefer-template': 'error',
     '@typescript-eslint/no-use-before-define': 'off',
     '@typescript-eslint/no-misused-promises': 'warn',
-    '@typescript-eslint/require-await': 'warn',
-    '@typescript-eslint/camelcase': 'warn',
+    '@typescript-eslint/require-await': 'off',
     '@typescript-eslint/unbound-method': 'off',
     '@typescript-eslint/no-empty-interface': 'off',
     '@typescript-eslint/explicit-function-return-type': ['warn', { allowExpressions: true }],
     '@typescript-eslint/no-unused-vars': ['warn', { argsIgnorePattern: '^_' }],
     '@typescript-eslint/array-type': ['error', { array: true }],
-    '@typescript-eslint/no-floating-promises': 'error'
+
+    '@typescript-eslint/explicit-module-boundary-types': [
+      'warn',
+      {
+        allowTypedFunctionExpressions: true,
+        allowHigherOrderFunctions: true,
+        allowDirectConstAssertionInArrowFunctions: true,
+        shouldTrackReferences: true,
+      },
+    ],
+    '@typescript-eslint/naming-convention': [
+      'warn',
+      { selector: 'default', format: ['camelCase'] },
+
+      { selector: 'variableLike', format: ['camelCase'] },
+      {
+        selector: 'variable',
+        format: ['camelCase', 'UPPER_CASE'],
+        leadingUnderscore: 'allow',
+      },
+      {
+        selector: 'parameter',
+        format: ['camelCase'],
+        leadingUnderscore: 'allow',
+      },
+
+      { selector: 'memberLike', format: ['camelCase'] },
+      {
+        selector: 'memberLike',
+        modifiers: ['private'],
+        format: ['camelCase'],
+        leadingUnderscore: 'require',
```

Comment:
Can we change it to 'allow'?
In my experience leading underscore is useful in some cases (eg setters, getters) but obstructs readability in other cases. So i would rather have the choice to use it or not depending on my usecase.

Reply:
Nice catch, I moved this config partly from my last project, so there may be some issues 😄